### PR TITLE
[SYCL][COMPAT] nd_range barriers seq_cst by default in supported devices

### PR DESCRIPTION
**Pull request by Alcpz**
AMD ~~and CUDA~~ devices still not supported.
~~CUDA to be supported in https://github.com/intel/llvm/pull/12516~~

Edit: Since #12516 has been merged, CUDA is also `seq_cst` by default.

### CODE REVIEW

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -1129,17 +1129,24 @@ spec, and so should be used with caution.
 namespace syclcompat {
 namespace experimental {
 
+#if defined(__AMDGPU__)
+// seq_cst currently not working for AMD
```
Comment (joeatodd) [inline sycl/doc/syclcompat/README.md:1133]: Do we need a convention to start tracking `TODO:`/`FIXME:` in our headers and/or docs? Or can we handle this with Jira?

**Review by joeatodd — APPROVED** [review]
All looks good :+1:  Depending what you think about the TODO thing, we should add that, or create a ticket. :ship:

File: sycl/doc/syclcompat/README.md (hunk 1 above)
Comment (Alcpz) [reply sycl/doc/syclcompat/README.md:1133]: Yes. As discussed, It's a good idea to have a `FIXME(@intel/syclcompat-lib-reviewers):` to track this kind of issues.

**Comment by Alcpz** [comment]
@intel/llvm-gatekeepers This is ready for merge. Thanks

**Pull request merged**